### DISCUSSION TITLE
How to deploy a code from specific branch

### DISCUSSION THREAD

**Discussion by irohitb** [vercel/community#3304] [Help] [answered]
### Summary

Hey, So I am working on a monorepo project divided into packages and apps. 

I have added an app which I want to deploy only from a branch. (let's call the app as `xyz` and branch as `feat/xyz`). 

Now, when I try to select a project and deploy it, it automatically selects branch to `main` which won't have this app (this app is only in `feat/xyz`)

### Steps to Reproduce

- Vercel Dashboard
- Add/New
- Select Project. 
- Select Apps (but branch is on main only)

**Comment by amyegan** [comment]
👋 Hey @irohitb. You can customize the [Production Branch](https://vercel.com/docs/concepts/deployments/git#production-branch) on the Git page in Project Settings.

**Comment by irohitb** [comment]
@amyegan once the project is deployed from that branch, is there a way I can make that project to not show up in other PR's? Since the branch is not merged,  CI fails as it can't find that project.

**Comment by swarnava** [comment]
> is there a way I can make that project to not show up in other PR's?

Hi, Could you clarify bit more?

**Comment by irohitb** [comment]
@swarnava Sorry, So I have project which have many `apps` (app1, app2, app3, app4) all of them are merged in main, all of them have their separate project in vercel and their separate branch as `source`. (production-app1, production-app2 etc)

I added a new project `app5`, this app5 isn't merged in`main`. When I push a commit , the `Vercel` inspect on github works fine on `production-app5` branch with Vercel deploy showing successful deployment for all the projects (app1, app2, app3). 

but If I push it to any other branch (feat/some-feature) then vercel inspect throws error that it can't find `app5` against which it should run deployment (since the other branches won't have app5 code). 

Does this make sense?

**Comment by swarnava** [comment, marked as answer]
Thank you! It sounds like you have a Monorepo which isn't setup correctly. You can review following instruction: https://vercel.com/docs/concepts/monorepos/. This will ensure the build happens only to specific apps when you commit to the repository.
